core/deployment/src/main/java/io/quarkus/deployment/ide/IdeProcessor.java

```diff
@@ -130,13 +132,26 @@ public IdeFileBuildItem detectIdeFiles(LaunchModeBuildItem launchModeBuildItem,
         if (launchModeBuildItem.getDevModeType().orElse(null) != DevModeType.LOCAL) {
             return null;
         }
+
         Set<Ide> result = new HashSet<>(2);
-        Path projectRoot = buildSystemTarget.getOutputDirectory().getParent();
-        IDE_MARKER_FILES.forEach((file, ides) -> {
-            if (Files.exists(projectRoot.resolve(file))) {
-                result.addAll(ides);
+        Path root = buildSystemTarget.getOutputDirectory();
+
+        // hack to try and guess the IDE when using a multi-module project
+        for (int i = 0; i < 3; i++) {
+            root = root.getParent();
+            for (Map.Entry<String, List<Ide>> entry : IDE_MARKER_FILES.entrySet()) {
+                String file = entry.getKey();
+                List<Ide> ides = entry.getValue();
+                if (Files.exists(root.resolve(file))) {
+                    result.addAll(ides);
+                }
```

Comment:
Have you tried this when for example a project is in `/tmp/someproject` and there is no `.idea` directory in `someproject`? 
My fear is that this will blow up will some obscure error

Reply:
In that case it will definitly blow up, getParent() returns null if the current path has no parent.
Will fix this later today.

Reply:
done